cmd/manager/main.go

```diff
@@ -19,26 +19,10 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
 )
 
-var log = logf.Log.WithName("cmd")
-
-func init() {
-	// Set log level... override default w/ command-line variable if set.
-	debugBool := shared.GetBoolEnv("DEBUG") // info, debug
-
-	// The logger instantiated here can be changed to any logger
-	// implementing the logr.Logger interface. This logger will
-	// be propagated through the whole operator, generating
-	// uniform and structured logs.
-	if debugBool {
-		logf.SetLogger(logf.ZapLogger(true))
-	} else {
-		logf.SetLogger(logf.ZapLogger(false))
-	}
-}
+var log = kieapp.SugaredLogger("cmd")
```

Comment:
My concern is that this is not our layer of indirection. It is "zap.SugaredLogger".

Here is a test for whether the layer of indirection solves the problem. You changed around 16 files in order to switch to the new logger. How many files would you need to change if we switched back to the old logger now? Ideally, the answer would be, only our own logger file.

Reply:
I had to change them to get away from the controller-runtime logger reference in every pkg

Reply:
going forward, it would only be changes in our logger class

Reply:
but ya, i can change the function name to something more generic?

Reply:
It's not about the function name. The interface you return is "zap.SugaredLogger", which may be ok. It just means that if we switch to another logger implementation, it will still be a lot of work. we would have to implement some 20+ functions and delegate to the new logger. I'm ok with crossing that bridge when we get to it, though.

Reply:
I've at least changed the function name to something more generic to make it easier going forward and ya, I'm sure can be improved. Let's get these sdk upgrades merged so we can iterate around logging and other items we have.